recipe/meta.yaml

```
@@ -0,0 +1,69 @@
+{% set name = "momepy" %}
+{% set version = "0.7.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/momepy-{{ version }}.tar.gz
+  sha256: 413ed9221dc4b2190621f8cb37789f09ea882f7e81dfbffd5731a22da6e61bab
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+  skip: True  # [py<310 or s390x]
+
```

@cbouss since this package defines the version in the same way as `libpysal` is doing, shall we also, in this case, tell `setuptools_scm` about the version?

```suggestion
  script_env:
    - SETUPTOOLS_SCM_PRETEND_VERSION={{ version }}

```

see:

https://github.com/pysal/momepy/blob/78c1b244f3acef60d77cbce98e3426c755b3033d/pyproject.toml#L9

https://github.com/pysal/momepy/blob/78c1b244f3acef60d77cbce98e3426c755b3033d/momepy/__init__.py#L20-L21

how can we test if `setuptools` got the version right?

---

Yes

---

`SETUPTOOLS_SCM_PRETEND_VERSION` is not needed because the version is defined correctly in the logs:

```
  Created wheel for momepy: filename=momepy-0.7.0-py3-none-any.whl size=277755 sha256=0b511371a513ceb95a5808e83002d695a2c2a94447391063920d941c4b740078

  Stored in directory: /tmp/pip-ephem-wheel-cache-2wm7lafp/wheels/72/18/a3/867e1df13a22456752599036d90910f6c361764d9fc176be19

Successfully built momepy

Installing collected packages: momepy
Successfully installed momepy-0.7.0
```

---

I checked the log of `libpysal` in the PR that generated the package with the wrong version `0.0.0` but it seems that the version was wrongly picked even if in the log I can see `4.10`:

https://github.com/AnacondaRecipes/libpysal-feedstock/commit/243f02c8cb9b6c13f00b2e64600f70bb1dfc515f

```
...
                   INFO     Preparing metadata (pyproject.toml): finished with status 'done'                                                                                                                         
                   INFO     Source in /private$SRC_DIR has version 4.10, which satisfies requirement libpysal==4.10 from file:///private$SRC_DIR                                                                     
                   INFO     Removed libpysal==4.10 from file:///private$SRC_DIR from build tracker '/private/tmp/pip-build-tracker-bhanaiq5'                                                                         
                   INFO   Created temporary directory: /private/tmp/pip-unpack-f6gjnh9u                                                                                                                              
                   INFO   Building wheels for collected packages: libpysal                                                                                                                                           
                   INFO     Created temporary directory: /private/tmp/pip-wheel-d1g5pk8e                                                                                                                             
                   INFO     Destination directory: /private/tmp/pip-wheel-d1g5pk8e                                                                                                                                   
                   INFO     Building wheel for libpysal (pyproject.toml): started                                                                                                                                    
                   INFO     Building wheel for libpysal (pyproject.toml): finished with status 'done'                                                                                                                
                   INFO     Created wheel for libpysal: filename=libpysal-4.10-py3-none-any.whl size=2793645 sha256=e88359700943f8378df5690e9f881821513cb89c498fe55e026da4cfd29351ac                                 
                   INFO     Stored in directory: /private/tmp/pip-ephem-wheel-cache-16l811wf/wheels/cc/f3/96/c03d210a52ab7161af8ed0b180dd9daa29fc0ce4dfc921b2aa                                                      
                   INFO   Successfully built libpysal                                                                                                                                                                
                   INFO   Installing collected packages: libpysal                                                                                                                                                    
                   INFO                                                                                                                                                                                              
                   INFO   Successfully installed libpysal-4.10                                                                                                                                                       
                   INFO   Removed build tracker: '/private/tmp/pip-build-tracker-bhanaiq5'                                                                                                                           
                   INFO                                                                                                                                                                                              
                   INFO   Resource usage statistics from building libpysal:                                                                                                                                          
                   INFO      Process count: 3                                                                                                                                                                        
                   INFO      CPU time: Sys=0:00:00.2, User=0:00:00.5                                                                                                                                                 
                   INFO      Memory: 69.7M                                                                                                                                                                           
                   INFO      Disk usage: 44.7K                                                                                                                                                                       
                   INFO      Time elapsed: 0:00:06.1                                                                                                                                                                 
                   INFO                                                                                                                                                                                              
                   INFO                                                                                                                                                                                              
                   INFO   Packaging libpysal                                                                                                                                                                         
                   INFO   Packaging libpysal-4.10-py311hca03da5_0                                                                                                                                                    
05/31/24 20:40:54  INFO   compiling .pyc files...                                                                                                                                                                    
...
```

---

also, why `pip check` in `libpysal`'s PR did not throw an error? Shouldn't `pip check` be checking also the version of the package that was just installed (besides its dependencies)?
```
pip install lippysal
pip check # <- should have failed?
```

---

another way we can use to test if `setuptools` got the version right?